core/src/main/scala/kafka/admin/TopicCommand.scala

```diff
@@ -742,10 +760,16 @@ object TopicCommand extends Logging {
       if (has(bootstrapServerOpt) == has(zkConnectOpt))
         throw new IllegalArgumentException("Only one of --bootstrap-server or --zookeeper must be specified")
 
+      if (has(topicIdOpt) && has(zkConnectOpt)) {
+        throw new IllegalArgumentException("--topic-id can used only with --bootstrap-server")
+      }
+
       if (!has(bootstrapServerOpt))
         CommandLineUtils.checkRequiredArgs(parser, options, zkConnectOpt)
-      if(has(describeOpt) && has(ifExistsOpt))
-        CommandLineUtils.checkRequiredArgs(parser, options, topicOpt)
```

Comment:
We have removed this check?

Reply:
Yes, when we use `--describe`, `--topic` is not mandatory. I add a `if(has(topicOpt) == has(topicIdOpt))` to ensure only one of `--topic` and `--topic-id` is used.